### ISSUE TITLE
[Bug]: Wait functions hangs when trying to get an element handle from a new tab opened with target='_blank'

### ISSUE THREAD

**Issue by FanghanHu** [puppeteer/puppeteer#11158]
### Minimal, reproducible example

```TypeScript
//chrome-for-testing v117.0.5938.149
//2 pages are hosted by live server in vscode (first test, now hosted on github pages)
//Page 1 (index.html)
/*
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Page 1</title>
</head>
<body>
    This is page 1
    <a href="./page2.html" target="_blank">Page 2</a>
</body>
</html>
*/

//Page 2 (page2.html)
/*
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Page 2</title>
</head>
<body>
    This is page 2
    <button id="button" onclick="this.innerText='Good'">
        click me
    </button>
</body>
</html>
*/

import puppeteer from "puppeteer";

(async () => {
    // Launch the browser and open a new blank page
    const browser = await puppeteer.launch(
        {headless: false}
        // {headless: 'new'}
    )
    let page = await browser.newPage();

    // Navigate the page to a URL
    await page.goto("https://fanghanhu.github.io/Puppeteer-Test/");
    // Set screen size
    await page.setViewport({ width: 1080, height: 1024 });

    const link = await page.locator("a").waitHandle();

    const pageTarget = page.target();

    const [newTarget] = await Promise.all([
        browser.waitForTarget((target) => target.opener() === pageTarget),
        link.click(),
    ]);

    page = await newTarget.page();

    console.log("before click");
    //const button = await page.locator('#button').waitHandle({timeout: 2000});
    const button = await page.waitForSelector("#button", {timeout: 2000});
    await button.click();
    console.log("After click");

    await browser.close();
})();
```

### Error string

no error

### Bug behavior

- [X] Flaky
- [ ] PDF

### Background

Trying to write some browser testing script, needed to test a button open the correct page in a new tab.

### Expectation

the button is clicked, and see "before click" and "After click" in the console.

### Reality

sometimes I only see "before click".
when trying to get an element handle from the newly opened tab, The waitForSelector function hangs randomly, (10% in `headless:'new'`, 20% in `headless:false`, 80% in `old headless mode`, tested 10 times for each mode)

instead of timing out after 2s and complaining about not finding the element, the script just hangs indefinitely until something else timeout.

Update: now hosting the 2 HTML files on Github Pages, the behavior has changed: when it hangs it no longer timeout by other stuff, it just hangs indefinitely. also old headless mode seems to not hang anymore while the other 2 modes still hang.

### Puppeteer version

21.3.8

### Node version

18.17.1

### Package manager

npm

### Package manager version

9.5.1

### Operating system

Windows

**Comment by jskorlol** [comment]
https://github.com/puppeteer/puppeteer/issues/11103
https://github.com/puppeteer/puppeteer/issues/11089

This issue is likely the same as the aforementioned one. It's been updated, but the release hasn't been deployed yet.

**Comment by FanghanHu** [comment]
I can confirm this issue is fixed in 2.4.1
